crates/sema/src/ast_passes.rs

```diff
@@ -123,6 +123,32 @@ impl<'ast> Visit<'ast> for AstValidator<'_> {
         self.walk_stmt(stmt)
     }
 
+    fn visit_item_contract(
+        &mut self,
+        contract: &'ast ast::ItemContract<'ast>,
+    ) -> ControlFlow<Self::BreakValue> {
+        let ast::ItemContract { kind: _, name, bases: _, body } = contract;
+        let contract_name = name.as_str();
+
+        for item in body.iter() {
+            if let ast::ItemKind::Function(ast::ItemFunction { kind: _, header, body: _ }) =
+                &item.kind
+            {
+                if let Some(func_name) = header.name {
+                    if func_name.as_str() == contract_name {
```

Comment:
```suggestion
                    if func_name == contract.name {
```

Compare Symbols not strings, it has the same result but faster, which is the point of Symbol in the first place